tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -714,6 +714,18 @@ def insights_property_test(self):
           False
         )
 
+    def subscription_attached_property_test(self):
+        """Test the IsSubscriptionAttached property."""
+        # should be false by default
+        self.assertFalse(self.subscription_interface.IsSubscriptionAttached)
+
+        # this property can't be set by client as it is set as the result of
+        # subscription attempts, so we need to call the internal module interface
+        self.subscription_module.set_subscription_attached(True)
+
+        # now the property should be True
+        self.assertTrue(self.subscription_interface.IsSubscriptionAttached)
```

Comment:
I think you should use `check_dbus_property` with a custom setter here. Otherwise you don't test the signal at all.

Reply:
Good point, did just that. :)